vizro-core/src/vizro/figures/kpi_cards.py

```diff
@@ -0,0 +1,167 @@
+    value_column: str,
+    reference_column: str,
+    *,
+    value_format: str = VALUE_FORMAT_DEFAULT,
+    reference_format: str = "{delta_relative:.1%} vs. reference ({reference})",
+    agg_func: str = "sum",
+    title: Optional[str] = None,
+    icon: Optional[str] = None,
+) -> dbc.Card:
+    """Creates a styled KPI (Key Performance Indicator) card displaying a value in comparison to a reference value.
+
+    Args:
+        data_frame: Data frame containing the data.
+        value_column: Column name of the value to be shown.
+        reference_column: Column name of the reference value for comparison.
+        value_format: Format string to be applied to the value. It must be a valid Python format string where any of the
+            below placeholders can be used. Defaults to "{value}".
+            - value: `value_column` aggregated by `agg_func`.
+            - reference: `reference_column` aggregated by `agg_func`.
+            - delta: Difference between `value` and `reference`.
+            - delta_relative: Relative difference between `value` and `reference`.
+
+            For more details see: https://docs.python.org/3/library/string.html#format-specification-mini-language
+
+            Common examples include:
+             - "{value}": Displays the raw value.
+             - "${value:0.2f}": Formats the value as a currency with two decimal places.
+             - "{value:.0%}": Formats the value as a percentage without decimal places.
+             - "{value:,}": Formats the value with comma as a thousands separator.
+
+        reference_format: Format string to be applied to the reference. For more details on possible placeholders, see
+            docstring on `value_format`. Defaults to "{delta_relative:.1%} vs. reference ({reference})".
+        agg_func: String function name to be used for aggregating the data. Common options include
+            "sum", "min", "max", "mean" or "median". Default is "sum".
+        title: KPI title displayed on top of the card. If not provided, it defaults to the capitalized `value_column`.
+        icon: Name of the icon from the [Google Material Icon Library](https://fonts.google.com/icons) to be displayed
+            on the left side of the KPI title. If not provided, no icon is displayed.
+
+    Raises:
+        UserWarning: If `value_format` or `reference_format` is provided, a warning is raised to make aware
+            that only trusted user input should be provided.
+
+    Returns:
+        A Dash Bootstrap Components card (`dbc.Card`) containing the formatted KPI value and reference.
+
+    """
+    if value_format != VALUE_FORMAT_DEFAULT or reference_format != REFERENCE_FORMAT_DEFAULT:
+        warnings.warn(FORMATTING_WARNING, UserWarning)
+
+    title = title or value_column.title()
+    value, reference = data_frame[[value_column, reference_column]].agg(agg_func)
+    delta = value - reference
+    delta_relative = np.nan if reference == 0 else delta / reference
```

Comment:
```suggestion
    delta_relative = delta / reference if reference else np.nan
```

Ternary statements read much more easily with the edge case in the "else" rather than the "if". Also no need to compare to `0` since it's just falsey.